### DISCUSSION TITLE
Suggestions for missing characters.

### DISCUSSION THREAD

**Discussion by Nibb31** [qwerty-fr/qwerty-fr#66] [General]
A couple of missing characters that are commonly used:
- ° (degree):  AltGr+0
- ring diacritic dead key (as in Å or ångstrom): AltGr+Shift+0
- µ (mu): AltGr+M
- ™ (trademark): AltGr+T
- ³ (exponent 3): AltGr+Shift+2
- ‰ (perthousand): AltGr+5
- §:AltGr+Shift+5

In addition, I suggest the following changes:
- ß and ẞ (capital) should be on AltGr+S and AltGr+Shift+S because it corresponds to a double S in German.
- € (euro) and £ (UK pound) should be on the AltGr+4 and AltGr+Shift+4 in order to have common currencies on the same key.

**Comment by Nibb31** [comment]
I'm going to add that the suggestion regarding the ß (eszett) and ẞ (capital eszett) being on the S key does break the ligature æ. So it would be a matter of prioritizing. 

Maybe ligatures could be moved to a dead key. That would also free up the P key for § (paragraph)
Or maybe they could be moved down to the X key (æ) and the the ; key (œ) (if the ° moves up to the 0 key).

**Comment by daks** [comment]
For ° (degree) it's already implemented just not represented in the 'Try it online' page. But if you look at https://github.com/qwerty-fr/qwerty-fr/blob/master/README.md you can find it on the :; key.
Not sure about the others.

**Reply by Nibb31** [reply]
Yes, I realize this is down to personal preference, but I personally find it more logical and easier to remember to have the ° on the 0 key. And the 'ring' diacritic (Å or å) could be Shift+AltGr+0.

This would free up the ;: key for the ligature œ, putting it in a similar position relative to O as the æ would be to A on the X key.

It seems logical to me, but I realize that it might not be for others.

**Comment by devnoname120** [comment]
IMO your suggestions are not well thought out and can be refuted easily. I don't mean to be rude, sorry if it comes across that way.

> ° (degree): AltGr+0

- That key is already occupied by <kbd>ø</kbd>. Where would we move it?
- Note: FYI `°` is already available on <kbd>AltGr</kbd> <kbd>Shift ⇧</kbd> <kbd>;</kbd> although not ideally accessible.

> ring diacritic dead key (as in Å or ångstrom): AltGr+Shift+0

- It's really hard to reach <kbd>AltGr</kbd> <kbd>Shift ⇧</kbd> <kbd>0</kbd> IMO. Currently it's used for `Ø` which is a character that is not very common and only in a few languages so the awkward location for that character is OK for me. If possible I prefer to keep `°` somewhere that is more easily accessible.

> µ (mu): AltGr+M

- <kbd>AltGr</kbd> <kbd>m</kbd> is already used to activate the math mode.
- It's already available in the greek mode: <kbd>AltGr</kbd> <kbd>g</kbd> then <kbd>m</kbd>.
- `μ` is a very rarely used letter and by very few people. Giving it a world-class position is not justified. 

> ™ (trademark): AltGr+T

- That key is already occupied by the Icelandic `þ`.
- Very few people need that symbol.

> ³ (exponent 3): AltGr+Shift+2

- It's already occupied by `˝` but I'm open to move that diacritic (I don't remember why I put it there).
- You can already do that with <kbd>AltGr</kbd> <kbd>6</kbd> then <kbd>3</kbd>.

> ‰ (perthousand): AltGr+5

- Almost nobody ever uses that, even among physicians and mathematicians.
- Where would we move the macron dead key `¯`? I'm open to moving it elsewhere because the location of this diacritic is a bit awkward, but definitely not for the `‰` symbol.

> §:AltGr+Shift+5

- Almost nobody uses that character AFAIK.
- What's the logic with putting it there?
